directory/cli/src/user.py

```diff
@@ -74,7 +74,93 @@ def show(login):
             error = '{}: user not found'.format(login)
             raise click.ClickException(error)
 
-    wrapper_commands = [
+    if not utils.advanced_mode_enabled():
```

Comment:
Defining these simple mode commands has got quite long and very nested - long term I think it would make sense to extract modules for both the simple and advanced commands, and just use these here in a similar way to how `user.add_commands` is called in `directory.py`. Alternatively, for now it might just be clearer to break the new simple mode commands up a bit. Just splitting out some helper function for the main parts of the `create`/`modify` commands would make it a lot clearer what's going on.

Extracting modules for the simple and advanced mode commands would also have the advantage of meaning we wouldn't need to always define all of the advanced mode commands below, only to have these ignored and unused when we're in simple mode.

Reply:
I think in the interest of saving time this can be put on hold until we have all of the base functionality sorted. After both the `user` and `group` commands are streamlined for simple mode I can come back and start doing things like this which I completely agree would be good for code maintainability.